timesketch/ui/graphs/graph-view.component.ts

```diff
@@ -0,0 +1,130 @@
+import {Component, Input} from '@angular/core'
+
+import {CytoscapeSettings} from './cytoscape-settings.component'
+
+export type GraphViewState = {
+  type: 'loading'
+} | {
+  type: 'empty'
+} | {
+  type: 'ready'
+  elements: Cy.ElementsDefinition | Cy.ElementDefinition[]
+}
+
+@Component({
+  selector: 'ts-graphs-graph-view',
+  templateUrl: './graph-view.component.html',
+})
+export class GraphViewComponent {
+  // tslint:disable-next-line:no-unused-variable
+  @Input() state: GraphViewState
+  settings: CytoscapeSettings = {
+    // interaction options:
+    minZoom: 1e-50,
+    maxZoom: 1e50,
+    zoomingEnabled: true,
+    userZoomingEnabled: true,
+    panningEnabled: true,
+    userPanningEnabled: true,
+    boxSelectionEnabled: true,
+    selectionType: 'single',
+    touchTapThreshold: 8,
+    desktopTapThreshold: 4,
+    autolock: false,
+    autoungrabify: false,
+    autounselectify: false,
+    // rendering options:
+    headless: false,
+    styleEnabled: true,
+    hideEdgesOnViewport: false,
+    hideLabelsOnViewport: false,
+    textureOnViewport: false,
+    motionBlur: false,
+    motionBlurOpacity: 0.2,
+    wheelSensitivity: 1,
+    pixelRatio: ('auto' as any),
+  }
+  layout = {
```

Comment:
Do we have an interface for layout as well?

Reply:
There is one in type definitions of Cytoscape library.

Reply:
I will it here, you are right.